server/service/src/sync/translation_and_integration.rs

```diff
@@ -57,12 +53,16 @@ impl<'a> TranslationAndIntegration<'a> {
                 SyncBufferAction::Merge => return Err(anyhow::anyhow!("Merge not implemented")),
             };
 
-            if let Some(records) = translation_result {
-                break Some(records);
+            if let Some(translation_result) = translation_result {
+                translation_results = translation_results.join(translation_result);
             }
-        };
+        }
 
-        Ok(result)
+        if translation_results.is_empty() {
```

Comment:
Can we just remove all optionals ? From translations (return empty vec ?), i still kind of like the optional in translators, it reads well  and don't have to initialise empty IntegrationRecords if no translations found

Reply:
I actually think we need four state of translation return: 'Translated', 'Ignored', 'Skipped', 'Error' (where ignored is deliberately ignored record, and skipped where translator doesn't match and error when something really bad happened, like name not found for name_store_join)

Reply:
might get tricky with more than one translator handling a record though...